### PR TITLE
WT-5597 Fix the history store file access during recovery

### PR DESCRIPTION
**Pull request by kommiharibabu**
Fix the scenarios where the history store file doesn't
exist in metadata during recovery.

### CODE REVIEW

**Comment by agorrod** [comment]
I had a conversation with Vamsi and Hari about this. There is a similar problem that is already resolved for non-history store tables (which involves renaming files out of the way and creating new files), but the solution to that results in an error message being reported to the console.

This approach avoids the error message being logged, but at the expense of having two different code paths handling the case where files exist on disk but not in the metadata table.

**Comment by kommiharibabu** [comment]
I can't think of any other way to use the existing approach without raising an error message as the file exists. I feel we may need to go with the approach in the patch for this history store.

File: src/txn/txn_ckpt.c [hunk 1]
```diff
@@ -850,23 +850,28 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
...
     /*
      * Get a history store dhandle. If the history store file is opened for a special operation this
-     * will return EBUSY which we treat as an error.
+     * will return EBUSY which we treat as an error. In scenarios where the history store is not
+     * part of the metadata file (performing recovery on backup folder where no checkpoint
+     * occurred), this will return ENOENT which we ignore and continue.
      */
-    WT_ERR(__wt_session_get_dhandle(session, WT_HS_URI, NULL, NULL, 0));
-    hs_dhandle = session->dhandle;
+    ret = __wt_session_get_dhandle(session, WT_HS_URI, NULL, NULL, 0);
```
Comment (agorrod) [inline src/txn/txn_ckpt.c]: Can you simplify the flow here, to have:

`WT_ERR_ERROR_OK(ret, ret = __wt_session_get_dhandle(session, WT_HS_URI, NULL, NULL, 0));`

Then remove the `if (ret == 0) {`?

File: src/txn/txn_ckpt.c [hunk 2]
```diff
@@ -850,23 +850,28 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
...
-    if (hs_dhandle != NULL) {
-        time_start_hs = __wt_clock(session);
-        WT_WITH_DHANDLE(session, hs_dhandle, ret = __wt_checkpoint(session, cfg));
-        WT_ERR(ret);
-        time_stop_hs = __wt_clock(session);
-        hs_ckpt_duration_usecs = WT_CLOCKDIFF_US(time_stop_hs, time_start_hs);
-        WT_STAT_CONN_SET(session, txn_hs_ckpt_duration, hs_ckpt_duration_usecs);
-    }
+        /*
+         * TODO: It is possible that we don't have a history store file in certain recovery
```
Comment (agorrod) [inline src/txn/txn_ckpt.c]: Is this still a todo?

File: src/txn/txn_ckpt.c (hunk 1 above)
Comment (kommiharibabu) [reply src/txn/txn_ckpt.c]: Fixed.

File: src/txn/txn_ckpt.c (hunk 2 above)
Comment (kommiharibabu) [reply src/txn/txn_ckpt.c]: I think NO. I didn't find any scenario where the `dhandle` is NULL. In error cases, the `dhandle` is not opened. The check is updated accordingly.

**Pull request merged**